akka-http-core/src/test/scala/akka/http/scaladsl/model/UriSpec.scala

```diff
@@ -500,7 +500,7 @@ class UriSpec extends WordSpec with Matchers {
 
       // empty host
       Uri("http://:8000/foo") shouldEqual Uri("http", Authority(Host.Empty, 8000), Path / "foo")
-      Uri("http://:80/foo") shouldEqual Uri("http", Authority(Host.Empty, 80), Path / "foo")
```

Comment:
Although I changed this line to pass the unit test, I did have a question about this case. Should we expect the port number of  
```scala
Uri("http", Authority(Host.Empty, 80), Path / "foo")
```
to be normalize . In Spary, this would be normalized. Due to this issue (https://github.com/akka/akka/issues/19650) and this PR (https://github.com/akka/akka/pull/20490), this would not be normalized. In https://github.com/akka/akka/pull/20490, it removed this https://github.com/akka/akka/blob/v2.4.4/akka-http-core/src/main/scala/akka/http/scaladsl/model/Uri.scala#L222. The only solution I came up is restoring this line then make a similar method but without normalizing the port number for the method `copy`.

Reply:
Thanks a lot, @jiminhsieh for going digging through the history. I think so far the changes are pretty straight-forward. So, let's merge this one and continue the discussion on a new PR/issue.

Reply:
Btw I also wonder whether these tests shouldn't actually return a different result:

https://github.com/akka/akka-http/blob/944e19475450afcbefbc7dac9914aa7d0470526f/akka-http-core/src/test/scala/akka/http/scaladsl/model/UriSpec.scala#L762-L765

i.e. partly revert the changes done back then.